src/main/kotlin/com/example/goodtripserver/places/PlacesController.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.goodtripserver.places
+
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import java.net.http.*
+import org.springframework.http.*
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.ResponseBody
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.util.UriComponentsBuilder
+import java.net.URI
+import java.net.http.HttpRequest
+
+@RestController
+class PlacesController {
+
+    private final val objectMapper = jacksonObjectMapper()
+
+    private fun getUrl(placeRequest: PlaceRequest): String {
+        val url = UriComponentsBuilder.fromHttpUrl("https://maps.googleapis.com/maps/api/place/nearbysearch/json")
+            .queryParam("location", placeRequest.location.replaceFirst('%', '?'))//smth here
+            .queryParam("radius", placeRequest.radius.toString())
+        if (!placeRequest.rankBy.isNullOrEmpty()) { //TODO глянуть, как приходят данные
+            url.queryParam("rankBy", placeRequest.rankBy)
+        }
+        placeRequest.type?.let {
+            url.queryParam("rankBy", placeRequest.rankBy)
+        }
+        return url.queryParam("key", "API_KEY")
+            .encode()
+            .toUriString().replace('?', '%')
+            .replaceFirst('%', '?')//TODO сделать менее бабайским и добавить поля нормально
+    }
```

Comment:
I'm more confused by the constant change of one character. Because there are some problems when creating the url